### PR TITLE
Small api changes

### PR DESCRIPTION
**Pull request by snim2**
Some small API changes needed by the ZS work.

### CODE REVIEW

File: src/lib/ast.rs [hunk 1]
```diff
@@ -254,6 +254,18 @@ impl<T: Clone + PartialEq, U: PartialEq + Copy> Arena<T, U> {
         false
     }
 
+    /// Return the id of the first leaf node in this arena, if there is one.
+    pub fn get_first_leaf(&self) -> Option<NodeId<U>> {
+        if self.is_empty() {
```
Comment (ltratt) [inline src/lib/ast.rs]: It would probably be more idiomatic to do something like:

```rust
if let Some(node) = self.root {
   ...
} else {
   None
}
```

**Comment by snim2** [comment]
I did not know about the `mut` in  `let Some(mut var) = val { ... } else { ... }` before. 

Ready to squash?

**Comment by ltratt** [comment]
Yep, please squash.

**Comment by snim2** [comment]
Squashed

**Pull request merged**